src/App.tsx

```diff
@@ -1,17 +1,146 @@
-import React from 'react';
-import './App.css';
-
-import users from './api/users';
+import React, { useState } from 'react';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { PreparedTodo } from './types/PreparedTodo';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import './App.scss';
 
 const App: React.FC = () => {
+  const users: User[] = usersFromServer;
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const defaultNewTodo = {
+    userId: 0,
+    title: '',
+    completed: false,
+  };
+
+  const [newTodo, setNewTodo] = useState({ ...defaultNewTodo });
+
+  const [isUserValid, setIsUserValid] = useState(true);
+  const [isTextValid, setIsTextValid] = useState(true);
+
+  const isNewTodoDataValid = () => {
+    let isDataValid = true;
+
+    if (newTodo.userId === 0) {
+      setIsUserValid(false);
+
+      isDataValid = false;
+    }
+
+    if (newTodo.title === '') {
+      setIsTextValid(false);
+
+      isDataValid = false;
+    }
+
+    return isDataValid;
+  };
+
+  const handleChange = (key: string, value: string | number) => {
+    setNewTodo(prevTodo => ({
+      ...prevTodo,
+      [key]: value,
+    }));
+
+    setIsTextValid(true);
+    setIsUserValid(true);
+  };
+
+  const handleAddClick = () => {
+    if (!isNewTodoDataValid()) {
+      return;
+    }
+
+    setTodos(prevTodos => [...prevTodos, {
+      id: prevTodos[prevTodos.length - 1].id + 1,
+      ...newTodo,
+    }]);
```

Comment:
```suggestion
   setTodos(prevTodos => ([
     ...prevTodos, 
    {
      id: prevTodos[prevTodos.length - 1].id + 1,
      ...newTodo,
    }
]));
```
It will be much clearer that way